### ISSUE TITLE
Write Kubernetes Events for errors during reconciliation

### ISSUE THREAD

**Issue by lfrancke** [stackabletech/issues#158]
We decided in our Architecture Meeting that we'd like to expose certain things also as Kubernetes events.

In this first step we'll just emit all Errors during reconciliation also as Kubernetes events.

Later we might extract a more fully-featured wrapper around logging into the operator framework.

**Comment by fhennig** [comment]
**TLDR** This issue is fine with me the way it is, I just wanted to write down some thoughts I had :)

I think I still have some thoughts that I maybe should've brought up in the meeting but didn't think quick enough.

Actually now it makes a lot of sense to me to somehow combine logging and the events. I'm just not entirely sure how we want to do that. The [Recorder](https://docs.rs/kube/latest/kube/runtime/events/struct.Recorder.html) should be created in the init function and contains a reference to the current object. So I think maybe wrapping this with something that we create that also does logging is actually a good idea and not too far fetched.

When we start a bunch of things at the same time and they depend on each other, like, let's say, a zookeeper is needed by something else. I think it makes sense that this other component instead waits for the zookeeper and doesn't raise an error. I think that would be a very good canditate to put into an event and maybe also into a status. Just have a status "Waiting" or "Pending" or "Provisioned" and then some reason/event where it says "Waiting for Zookeeper Cluster to become ready" or something like that. So I wouldn't quit there on an error.

**Comment by lfrancke** [comment]
The Clickhouse Operator has this announcer package: https://github.com/Altinity/clickhouse-operator/blob/master/pkg/controller/chi/announcer.go

**Comment by hikhvar** [comment]
@fhennig in our internal operators we switched to kubernetes upstream conditions type to express the states of our resources:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

**Comment by teozkr** [comment]
This is blocked on https://github.com/kube-rs/kube-rs/issues/812.
